src/panels/ShareModelPanel/ShareModel.tsx

```diff
@@ -1,22 +1,89 @@
+import { useState, useEffect } from "react";
+import { useStore, useSelector, useDispatch } from "react-redux";
 import { useParams } from "react-router-dom";
+import { Formik, Field, Form } from "formik";
+import cloneDeep from "clone-deep";
 import useModelStatus from "hooks/useModelStatus";
 import { formatFriendlyDateToNow } from "app/utils/utils";
+import { setModelSharingPermissions } from "juju";
+
+import { getModelControllerDataByUUID } from "app/selectors";
 
 import Aside from "components/Aside/Aside";
 import PanelHeader from "components/PanelHeader/PanelHeader";
+import FormikFormData from "components/FormikFormData/FormikFormData";
 
 import type { EntityDetailsRoute } from "components/Routes/Routes";
 import type { TSFixMe } from "types";
 
 import "./share-model.scss";
 
+type modelControllerData = {
+  additionalController: undefined;
+  path: string;
+  url: string;
+  uuid: string;
+  version: string;
+};
+
+type user = {
+  user: string;
+  "display-name": string;
+  "last-connection": string | null;
+  access: string;
+};
+
+type usersAccess = {
+  [key: string]: string;
+};
+
+type userAccess = {
+  name: string;
+  access: string | null;
+};
+
+type DefaultRootState = {};
+
 export default function ShareModel() {
   const { modelName } = useParams<EntityDetailsRoute>();
+  const dispatch = useDispatch();
+  const store = useStore();
+
+  const [errorMsg, setErrorMsg] = useState<string | null>(null);
+  const [usersAccess, setUsersAccess] = useState<usersAccess>({});
+  const [newUserFormSubmitActive, setNewUserFormSubmitActive] = useState(false);
 
   const modelStatusData: TSFixMe = useModelStatus() || null;
 
+  const controllerUUID = modelStatusData?.info?.["controller-uuid"];
+  const modelUUID = modelStatusData?.info.uuid;
+
+  const modelControllerDataByUUID = getModelControllerDataByUUID(
+    controllerUUID
+  );
+
+  const modelControllerData: modelControllerData = useSelector(
+    modelControllerDataByUUID as (
+      state: DefaultRootState
+    ) => modelControllerData
+  );
+
+  const modelControllerURL = modelControllerData?.url;
   const users = modelStatusData?.info?.users;
 
+  useEffect(() => {
+    const clonedUserAccess: usersAccess | null = cloneDeep(usersAccess);
+
+    users?.forEach((user: user) => {
+      const displayName = user["user"];
+
+      if (clonedUserAccess) {
+        clonedUserAccess[displayName] = user?.["access"];
+        setUsersAccess(clonedUserAccess);
+      }
+    });
+  }, [users]); // eslint-disable-line
```

Comment:
typically you should put the specific rule you're ignoring so that it'll still check the others.